packages/dds/tree/src/util/typeCheck.ts

```diff
@@ -127,13 +127,15 @@ export interface Invariant<in out T> extends Contravariant<T>, Covariant<T> {}
  * Compile time assert that X is True.
  * To use, simply define a type:
  * `type _check = requireTrue<your type check>;`
+ * @internal
```

Comment:
Not for this PR, but now that we have usages of these external to `tree`, we might consider moving these to a more general location.